File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/ModulesViewModels/SalesDemo/ImageHyperlinkBehavior.cs
using DevExpress.Mvvm.UI.Interactivity;
using System.Diagnostics;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Input;

namespace ProductsDemo {
    public class ImageHyperlinkBehavior : Behavior<Image> {
        public static readonly DependencyProperty NavigatetUriProperty =
            DependencyProperty.Register("NavigatetUri", typeof(string), typeof(ImageHyperlinkBehavior), new PropertyMetadata(null));
        public string NavigatetUri {
            get { return (string)GetValue(NavigatetUriProperty); }
            set { SetValue(NavigatetUriProperty, value); }
        }
        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.MouseLeftButtonUp += OnImageMouseLeftButtonUp;
        }
        protected override void OnDetaching() {
            AssociatedObject.MouseLeftButtonUp -= OnImageMouseLeftButtonUp;
            base.OnDetaching();
        }
        void OnImageMouseLeftButtonUp(object sender, MouseButtonEventArgs e) {
            Process.Start(new ProcessStartInfo { FileName = NavigatetUri.ToString(), UseShellExecute = true });
        }
    }
}

File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/ModulesViewModels/SalesDemo/ImageHyperlinkBehavior.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports System.Diagnostics
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Input

Namespace ProductsDemo

    Public Class ImageHyperlinkBehavior
        Inherits Behavior(Of Image)

        Public Shared ReadOnly NavigatetUriProperty As DependencyProperty = DependencyProperty.Register("NavigatetUri", GetType(String), GetType(ImageHyperlinkBehavior), New PropertyMetadata(CType(Nothing, PropertyChangedCallback)))

        Public Property NavigatetUri As String
            Get
                Return CStr(GetValue(NavigatetUriProperty))
            End Get

            Set(ByVal value As String)
                SetValue(NavigatetUriProperty, value)
            End Set
        End Property

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.MouseLeftButtonUp, AddressOf OnImageMouseLeftButtonUp
        End Sub

        Protected Overrides Sub OnDetaching()
            RemoveHandler AssociatedObject.MouseLeftButtonUp, AddressOf OnImageMouseLeftButtonUp
            MyBase.OnDetaching()
        End Sub

        Private Sub OnImageMouseLeftButtonUp(ByVal sender As Object, ByVal e As MouseButtonEventArgs)
            Call Process.Start(New ProcessStartInfo With {.FileName = NavigatetUri.ToString(), .UseShellExecute = True})
        End Sub
    End Class
End Namespace
